dkc/core/models/folder.py

```diff
@@ -82,6 +84,23 @@ def ancestors(self) -> models.query.RawQuerySet[Folder]:
             [self.pk, Folder.MAX_TREE_HEIGHT + 1],
         )
 
+    def increment_size(self, amount: int) -> None:
```

Comment:
You removed the transaction decorator from this method, was that intentional?

Reply:
Yeah, I was about to comment on that too. I think it's fine as-is, but I'd like some critical analysts.

The update itself is necessarily going to happen in it's own transaction, since it's a single statement. Until we have folder moves, the list of ancestor PKs will never change. Any subfolders added concurrently will be initialized with 0 size regardless.

It's possible now that between the update and the model reload, the size changes further in the database due to concurrent operations, but is that a problem? The new size would not be wrong, it's just more up-to-date than we'd expect. I don't think that's a problem for later local inconsistencies, since any other local models are outdated anyway.

Reply:
> The update itself is necessarily going to happen in it's own transaction, since it's a single statement.

That's not true, it can be a part of a larger transaction. In fact, the behavior we probably want here is `SELECT FOR UPDATE` inside of a transaction block, then the update query. Django has `select_for_update`, but I'm not sure it could be applied to the RawQuerySet returned from `self.ancestors` (I would assume it cannot.)

> It's possible now that between the update and the model reload, the size changes further in the database due to concurrent operations, but is that a problem? The new size would not be wrong, it's just more up-to-date than we'd expect. I don't think that's a problem for later local inconsistencies, since any other local models are outdated anyway.

No, this isn't a problem.

Reply:
One thing we could do is just add `FOR UPDATE` into our `ancestors` query. It would only hold onto locks while part of a transaction, and that would probably only occur if someone was making an update to the ancestors.

Reply:
For what it's worth, [from the Django docs](https://docs.djangoproject.com/en/3.1/topics/db/transactions/#why-django-uses-autocommit):
> In the SQL standards, each SQL query starts a transaction, unless one is already active. Such transactions must then be explicitly committed or rolled back.
> 
> This isn’t always convenient for application developers. To alleviate this problem, most databases provide an autocommit mode. When autocommit is turned on and no transaction is active, each SQL query gets wrapped in its own transaction. In other words, not only does each such query start a transaction, but the transaction also gets automatically committed or rolled back, depending on whether the query succeeded.

Reply:
I agree, when we consider folder move, we'd want to `SELECT FOR UPDATE` the ancestor rows within an explicit transaction which we open for the scope of the increment method. Until then, is there a need for a larger-scope transaction to tie these two queries (select the ancestor and update the sizes) together?

Reply:
I think it's fine for now.